### PR TITLE
Emphasise `variable_start_string`/`variable_end_string` affects templated vars

### LINKED ISSUE #80996: Emphasise `variable_start_string`/`variable_end_string` affects templated `vars`
### Summary

I want to template a file that is already a Jinja2 template. This necessitates configuring `variable_start_string`, `variable_end_string` etc. However, it turns out setting these also affects evaluation of templated variables defined in the same step. It is necessary to use the new marker strings for any templated variables. For example:

```yaml
---
# ...
- hosts: bastion
  tasks:
  # ...
  - name: Copy globals.yml file
    ansible.builtin.copy:
      src: files/globals.yml
    ansible.builtin.template:
      src: templates/globals.yml.j2
      dest: /etc/kolla/globals.yml
      block_start_string: "<%"
      block_end_string: "%>"
      variable_start_string: "<<"
      variable_end_string: ">>"
    vars:
      internal_vip_address: "<< hostvars['host-00']['ansible_host'] >>"
  # ...
```

This was previously reported as https://github.com/ansible/ansible/issues/43886, however, that was closed as the issue could not be fixed. This is understandable, however, there's no reason we shouldn't document it.

### Issue Type

Documentation Report

### Component Name

lib/ansible/modules/template.py

### Ansible Version

```console
$ ansible --version
ansible [core 2.14.2]
  config file = None
  configured module search path = ['/home/cloud/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /etc/kolla/venv/lib/python3.10/site-packages/ansible
  ansible collection location = /home/cloud/.ansible/collections:/usr/share/ansible/collections
  executable location = /etc/kolla/venv/bin/ansible
  python version = 3.10.6 (main, May 29 2023, 11:10:38) [GCC 11.3.0] (/etc/kolla/venv/bin/python3)
  jinja version = 3.1.2
  libyaml = True
```

### Configuration

```console
# if using a version older than ansible-core 2.12 you should omit the '-t all'
$ ansible-config dump --only-changed -t all
CONFIG_FILE() = None
```

### OS / Environment

Ubuntu 22.04

### Additional Information

The documentation should emphasise that the `variable_start_string`, `variable_end_string` etc. settings affects templated `vars`.

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

### PR DESCRIPTION
**Pull request by stephenfin**
##### SUMMARY

Fixes #80996

##### ISSUE TYPE

- Docs Pull Request

##### COMPONENT NAME

- `lib/ansible/modules/template.py`

##### ADDITIONAL INFORMATION

None.

### CODE REVIEW

File: lib/ansible/plugins/doc_fragments/template_common.py [hunk 1]
```diff
@@ -48,35 +48,43 @@ class ModuleDocFragment(object):
   block_start_string:
     description:
     - The string marking the beginning of a block.
+    - This will affect evaluation of templated variables in the same step.
```
Comment (bcoca) [inline lib/ansible/plugins/doc_fragments/template_common.py:51]: i would just add this once, to the `notes:` section

File: lib/ansible/modules/template.py [hunk 2]
```diff
@@ -79,6 +79,17 @@
...
+- name: Template a file using different marker strings
+  ansible.builtin.template:
+    src: globals.yml.j2
+    dest: /etc/kolla/globals.yml
+    block_start_string: "<%"
+    block_end_string: "%>"
+    variable_start_string: "<<"
+    variable_end_string: ">>"
+  vars:
+    vip_address: "<< hostvars['controller-00']['ansible_host'] >>"
```
Comment (bcoca) [inline lib/ansible/modules/template.py:91]: this is NOT limited to vars defined in task, but to ANY vars used in the template, no matter where they are defined.

**Comment by sivel** [comment]
I think we may be starting to question this behavior.  I might recommend holding off on merging this PR until we can decide whether we want to change the behavior to align with the `#jinja2` header.

**Comment by sivel** [comment]
I'm going to close this as resolved by https://github.com/ansible/ansible/pull/81005

That PR changes the behavior so that this arguments only affect the template itself and not dependent variables.

**Comment by stephenfin** [comment]
Awesome. Thank you! :pray:

**Pull request closed without merging**